src/UI/Art/SpriteArtist/SpriteUIUtils.java

```diff
@@ -25,7 +33,111 @@ public static double calculateSlideAmount(double currentNum, int targetNum)
+      int vBuffer)
+  {
+    ArrayList<String> items = new ArrayList<String>();
+    items.add(item);
+    drawTextMenu(g, bg, frame, bg, items, 0, mapX, mapY, hBuffer, vBuffer);
+  }
+  public static void drawBasicTextMenu(Graphics g, ArrayList<String> items, int selection, double mapX, double mapY)
+  {
+    drawTextMenu(g, SpriteMenuArtist.MENUBGCOLOR, SpriteMenuArtist.MENUFRAMECOLOR, SpriteMenuArtist.MENUHIGHLIGHTCOLOR, items,
+        selection, mapX, mapY, SpriteMenuArtist.menuHBuffer, SpriteMenuArtist.menuVBuffer);
+  }
+  public static void drawTextMenu(Graphics g, Color bg, Color frame, Color focus, ArrayList<String> items, int selection,
+      double mapX, double mapY, int hBuffer, int vBuffer)
+  {
+    {
+      XYCoord visualOrigin = SpriteMapView.getVisualOrigin();
+      // Find the dimensions of the menu we are drawing.
+      int menuWidth = getMenuTextWidthPx(items) + hBuffer * 2;
+      int menuHeight = getMenuTextHeightPx(items) + vBuffer * 2;
+
+      // Center the menu over the current action target location, accounting for the position of the map view.
+      int viewTileSize = getTileSize(); // Grab this value for convenience.
+      int drawX = (int) (mapX * viewTileSize - (menuWidth / 2 - viewTileSize / 2));// - origin.xCoord;
+      int drawY = (int) (mapY * viewTileSize - (menuHeight / 2 - viewTileSize / 2));// - origin.yCoord;
+
+      // Make sure the menu is fully contained in viewable space.
+      Dimension dims = SpriteOptions.getScreenDimensions();
+      drawX = (drawX < visualOrigin.xCoord) ? visualOrigin.xCoord
+          : (drawX > (visualOrigin.xCoord + dims.width - menuWidth)) ? (visualOrigin.xCoord + dims.width - menuWidth) : drawX;
+      drawY = (drawY < visualOrigin.yCoord) ? visualOrigin.yCoord
+          : (drawY > (visualOrigin.yCoord + dims.height - menuHeight)) ? (visualOrigin.yCoord + dims.height - menuHeight) : drawY;
+
+      // Draw the nice box for our text.
+      drawMenuFrame(g, bg, frame, drawX, drawY, menuWidth, menuHeight, vBuffer);
+
+      // Draw the highlight for the currently-selected option.
+      // selY = drawY plus upper menu-frame buffer, plus (letter height, plus 1px-buffer, times number of options).
+      int selY = drawY + vBuffer + (SpriteMenuArtist.menuTextHeight + SpriteOptions.getDrawScale()) * selection;
+      g.setColor(focus);
+      g.fillRect(drawX, selY, menuWidth, SpriteMenuArtist.menuTextHeight);
+
+      int drawScale = SpriteOptions.getDrawScale();
+      // Draw the actual menu text.
+      for( int txtY = drawY + vBuffer, i = 0; i < items.size(); ++i, txtY += SpriteMenuArtist.menuTextHeight + drawScale )
+      {
+        SpriteLibrary.drawTextSmallCaps(g, items.get(i), drawX + hBuffer, txtY, drawScale);
+      }
+    }
+  }
+
+  /**
+   * Populate 'out' with the string versions of the options available through 'menu'.
+   * @param menu
+   * @param out
+   */
+  public static void getMenuStrings(InGameMenu<? extends Object> menu, ArrayList<String> out)
```

Comment:
This function used to be a private member function of `SpriteMenuArtist`, which is still the only class that uses it. I'd be inclined to keep this over there unless there's a good reason to make it public/static.